### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -37,12 +49,19 @@ impl Error {
         }
     }
 
+    /// Does nothing if self.pointer is NULL; if self.pointer is non-NULL, then *self.pointer must be NULL.
+    /// A new GError is created and assigned to *self.pointer .
```
Comment (gkoz) [inline src/error.rs]: Discussions of pointers are likely to confuse. Unfortunately, simply copypasting the documentation doesn't always make sense.

File: src/date.rs [hunk 2]
```diff
@@ -81,6 +81,8 @@ pub struct Date {
 }
 
 impl Date {
+    /// Allocates a GDate and initializes it to a sane state. The new date will be cleared
```
Comment (gkoz) [inline src/date.rs]: Just an observation: ideally these types would not exist and the foreign ones would be converted to/from one of the Rust date/time types.

Comment (GuillaumeGomez) [inline src/date.rs]: For now they do exist, so I guess it's better (for now) to keep them.

File: src/error.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/error.rs]: I didn't but this one (and certainly others) escaped from my attention.

**Pull request merged**